src/uint256.cpp

```diff
@@ -18,39 +18,19 @@ std::string base_blob<BITS>::GetHex() const
 }
 
 template <unsigned int BITS>
-void base_blob<BITS>::SetHex(const char* psz)
+void base_blob<BITS>::SetHex(std::string_view str)
 {
-    std::fill(m_data.begin(), m_data.end(), 0);
-
-    // skip leading spaces
-    while (IsSpace(*psz))
-        psz++;
-
-    // skip 0x
-    if (psz[0] == '0' && ToLower(psz[1]) == 'x')
-        psz += 2;
-
-    // hex string to uint
-    size_t digits = 0;
-    while (::HexDigit(psz[digits]) != -1)
-        digits++;
-    unsigned char* p1 = m_data.data();
-    unsigned char* pend = p1 + WIDTH;
-    while (digits > 0 && p1 < pend) {
-        *p1 = ::HexDigit(psz[--digits]);
-        if (digits > 0) {
-            *p1 |= ((unsigned char)::HexDigit(psz[--digits]) << 4);
-            p1++;
-        }
+    const auto trimmed = util::RemovePrefixView(util::TrimStringView(str), "0x");
+    // Note: if we are passed a greater number of digits than would fit as bytes
+    // in m_data, we will be discarding the leftmost ones.
+    // str="12bc" in a WIDTH=1 m_data => m_data[] == "\0xbc", not "0x12".
+    size_t digits = std::ranges::find_if(trimmed, [](char c) { return ::HexDigit(c) == -1; }) - trimmed.begin();
```

Comment:
> Not sure about changing code that will be obsolete in a few commits. It seems wasteful to waste review on code that will be changed or deleted down the line. It seems better to touch every line of code only once (or the least number of times possible), than to repeatedly refactor it, and then remove the "feature".

The function is not being removed as of #30482.

> For example, by removing the `std::fill(m_data.begin(), m_data.end(), 0);`, two `SetHex` calls now may result in a different value, because the previous value is not zeroed, but leaked into the new value.

This is not true, the final loop iterates over the whole of `m_data`, filling zeroes on the remaining elements when running out of `digits`. It felt much more robust to me using a range-for, and cleaner then the previous `p1`/`pend`.